cases/scenarios/xstate-5/setup.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+set -exo pipefail
+
+cd "${0%/*}"
+SCENARIO_NAME=$(basename $PWD)
+SOLUTION_DIR=../../solutions/$SCENARIO_NAME
+
+if ! test -d $SOLUTION_DIR; then
+  git clone --filter blob:none https://github.com/statelyai/xstate.git $SOLUTION_DIR
+fi
+
+cd $SOLUTION_DIR
+git clean -fdx
+git reset --hard HEAD
+git switch --detach 6113a590a6e16c22f3d3e288b825975032bdfd48
```

Comment:
How should we maintain this in the future? From our PoV, it would be best to monitor the types performance continuously and this - again - pins this scenario to a specific version (I understand that this is better for your benchmarks :P)

Reply:
We can leave it unpinned, but it becomes useless as a long term benchmark. I'm not so sure which philosophy to have.

Reply:
We could also update it periodically or something. It's not like we are introducing big type changes with each commit.

> I'm not so sure which philosophy to have.

Yeaah, me neither 😅 How hard would it be to set up this infra on our CI to have it run per our PR? I remember that you mentioned that GHA are not perfect for benchmark stability... but kinda better than nothing, right? ;p It could help us to avoid catastrophic changes.

Reply:
It shouldn't be too hard, but you're right that the results will probably not be good. Variance on a single GHA machine is like +/-5%, which is very bad, and the machine specs are different each run. Our tuned perf machines are more like +/-0.3% and are stable.

Reply:
The thing we settled on was to have both pinned and unpinned variants. PRs will run unpinned code, while we have pinned variants for long term graphs. If we ever need to update a pinned benchmark, we will create a new benchmark and rename it (ending the line of the old one) and then backfill the data a bit if desired. See: #12

I'll update this PR when I have it settled and #14 is in.